src/scripts/main.js

```diff
@@ -1,3 +1,35 @@
 'use strict';
 
-// write code here
+function getEmployees() {
+  const list = document.querySelector('ul');
+
+  return [...list.querySelectorAll('li')].map((employee) => {
+    return {
+      name: employee.textContent.trim(),
+      position: employee.dataset.position,
+      salary: +employee.dataset.salary.slice(1).replaceAll(',', ''),
+      age: +employee.dataset.age,
+    };
+  });
+}
+
+function sortList(employees) {
+  return [...employees].sort(
+    (employeeA, employeeB) => employeeB.salary - employeeA.salary,
+  );
+}
+
+function updateList(sortedEmployees) {
+  const list = document.querySelector('ul');
+  const employees = [...list.querySelectorAll('li')];
+
+  for (let i = sortedEmployees.length - 1; i >= 0; i--) {
+    const employee = employees.find(
+      (li) => li.textContent.trim() === sortedEmployees[i].name,
+    );
+
+    list.prepend(employee);
```

Comment:
There is a potential issue here: if `employees.find(...)` returns `undefined` (i.e., if an employee is not found), `list.prepend(employee)` will throw an error. Consider adding a check to ensure `employee` is not `undefined` before calling `list.prepend(employee)`.